src/fiber/gossip.rs

```diff
@@ -0,0 +1,2822 @@
+            Ok((subscription, output_port)) => {
+                output_port.subscribe(receiver, converter);
+                Ok(subscription)
+            }
+            Err(e) => Err(Error::InternalError(anyhow::anyhow!(e.to_string()))),
+        }
+    }
+
+    async fn update_subscription(
+        &self,
+        subscription: &Self::Subscription,
+        cursor: Option<Cursor>,
+    ) -> Result<(), Self::Error> {
+        const DEFAULT_TIMEOUT: u64 = Duration::from_secs(5).as_millis() as u64;
+        call_t!(
+            &self.actor,
+            ExtendedGossipMessageStoreMessage::UpdateSubscription,
+            DEFAULT_TIMEOUT,
+            *subscription,
+            Some(cursor)
+        )
+        .map_err(|e| Error::InternalError(anyhow::anyhow!(e.to_string())))
+    }
+
+    async fn unsubscribe(&self, subscription: &Self::Subscription) -> Result<(), Self::Error> {
+        const DEFAULT_TIMEOUT: u64 = Duration::from_secs(5).as_millis() as u64;
+        call_t!(
+            &self.actor,
+            ExtendedGossipMessageStoreMessage::UpdateSubscription,
+            DEFAULT_TIMEOUT,
+            *subscription,
+            None
+        )
+        .map_err(|e| Error::InternalError(anyhow::anyhow!(e.to_string())))
+    }
+}
+
+struct BroadcastMessageOutput {
+    // The id of the subscriber. Mostly for debugging.
+    id: u64,
+    // This is the last cursor of the ExtendedGossipMessageStore when the subscriber is created.
+    // We have to send all messages up to this cursor to the subscriber immediately after the
+    // subscription is created. Other messages (messages after this cursor) are automatically
+    // forwarded to the subscriber by the store actor.
+    store_last_cursor_while_starting: Cursor,
+    // The filter that a subscriber has set. We will only send messages that are newer than this filter.
+    // This is normally a cursor that the subscriber is confident that it has received all the messages
+    // before this cursor.
+    filter: Option<Cursor>,
+    // A port that from which the subscriber will receive messages and from which we will send messages to the subscriber.
+    output_port: Arc<OutputPort<GossipMessageUpdates>>,
+}
+
+impl BroadcastMessageOutput {
+    fn new(
+        id: u64,
+        store_last_cursor_while_starting: Cursor,
+        filter: Option<Cursor>,
+        output_port: Arc<OutputPort<GossipMessageUpdates>>,
+    ) -> Self {
+        Self {
+            id,
+            store_last_cursor_while_starting,
+            filter,
+            output_port,
+        }
+    }
+}
+
+// This is the error type for processing the gossip messages.
+// Sometimes we can't be very sure of if a gossip message is valid on a first sight.
+// For example, a ChannelUpdate message may be signed with a different node's key,
+// but we won't be able to find this out until we have the corresponding ChannelAnnouncement.
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum GossipMessageProcessingError {
```

Comment:
it looks like variants `ProcessingError` and `NewerMessageSaved` are never constructed, could you point out the usage?